tests/parsec-cli-tests.sh

```diff
@@ -215,6 +217,27 @@ test_csr() {
     delete_key $1 $KEY
 }
 
+test_rsa_key_bits() {
+    KEY="anta-key-rsa-bits"
+    DEFAULT_SIZE=2048
+    
+    if [ -n "$1" ]; then
+       key_size=$1
+       key_param="--bits $1"
+    else
+       key_size=${DEFAULT_SIZE}
+       key_param=""
+    fi
+    
+    run_cmd $PARSEC_TOOL_CMD create-rsa-key --key-name $KEY $key_param
+    run_cmd $PARSEC_TOOL_CMD export-public-key --key-name $KEY >${MY_TMP}/checksize-${KEY}.pem
+    if ! run_cmd $OPENSSL rsa -pubin -text -noout -in ${MY_TMP}/checksize-${KEY}.pem | grep -q "RSA Public-Key: (${key_size} bit)"; then
+       echo "Error: create-rsa-key should have produced a ${key_size}-bit RSA key."
+       EXIT_CODE=$(($EXIT_CODE+1))
```

Comment:
I know it's not introduced in this change, but I wanted to ask why the script returns the EXIT_CODE this ways
if multiple tests fail then the EXIT_CODE would be hard to interpret, or is there something I don't understand?

Reply:
The idea is that the return code is equal to number of failed commands

Reply:
> I know it's not introduced in this change, but I wanted to ask why the script returns the EXIT_CODE this ways
if multiple tests fail then the EXIT_CODE would be hard to interpret, or is there something I don't understand?

I think this was a convention started by @anta5010 so I'll defer to him on that one. One observation I would make, though, is that this script is gradually evolving into the kind of integration test runner that might be better suited to a proper BDD tool as opposed to just a bash script.

Reply:
I agree. When I started this script the idea was "we need something simple for Parsec deployments which we can use to check that Parsec is functional with required providers". Now it's getting close to a command line test tool for all use cases. But, may be it's OK as we run this script as a part of Parsec CI pipeline as well.